R/OsModel.R

```diff
@@ -0,0 +1,127 @@
+#' Title - TODO
+#'
+#' Description - TODO
+#'
+#' @slot stan TODO
+#' @slot templated Logical. Defines whether the OsModel object needs to be completed
+#' @export
+OsModel <- setClass(
+    Class = "OsModel",
+    representation = list(
+        stan = "StanModule",
+        templated = "logical"
+    )
+)
+
+
+
+#' LogLogisticModule
+#'
+#' Log logistic module helper function.
+#' @param functions
+#' @param data
+#' @param parameters
+#' @param transformed_parameters
+#' @param generated_quantities
+#' @param priors
+#' @param inits
+#' @param templated
+#' @export
+LogLogisticModule <- function(functions = "os_functions.stan",
+                              data = "os_data.stan",
+                              parameters = "os_parameters.stan",
+                              transformed_parameters = "os_transformed_parameters.stan",
+                              generated_quantities = "os_generated_quantities.stan",
+                              priors = os_prior(),
+                              inits = list(),
+                              templated = TRUE) {
+    st_mod <- StanModule(
+            functions = functions,
+            data = data,
+            parameters = parameters,
+            transformed_parameters = transformed_parameters,
+            generated_quantities = generated_quantities,
+            priors = priors,
+            inits = inits
+        )
+
+    OsModel(
+        stan = st_mod,
+        templated = templated
+    )
```

Comment:
Ahh with regards to my above comment I think this function should just return the `StanModule` object not the OSModel object. This is just a help function for getting the correct stanmodule.  It does not need to be concerned with actually creating an OSModel

Reply:
So if the `LogLogisticModule` produces a `StanModule` with default options for the Log-Logistic overall survival submodel then there is no need for the `LogLogisticOs` object creator, is that right ?

Reply:
I think we clarified this over the standup but just to clarify  that I would expect  `LogLogisticModule` to return a `StanModule` and then this would be the default input to the `LogLogisticOs` constructor